test/core_test.exs

```diff
@@ -5,6 +5,11 @@ defmodule CoreTest do
   import TelemetryMetricsPrometheus.Core, only: [init: 2, scrape: 1, stop: 1]
   alias Telemetry.Metrics
 
+  test "has a child spec" do
+    assert %{id: TelemetryMetricsPrometheus.Core.Registry} =
```

Comment:
Building on the other child spec comments, let's also make sure this fully tests the child spec generation with all options, e.g. giving a name should set the id.

Another thing is we can exercise testing some of this by eliminating the calls to `init` and `stop` in all of these tests now since we can start this supervised now! We can replace those with `start_supervised` in each test. :)

Let's keep one test to `init` just to make sure we haven't broken anything.